homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,413 @@
+        self._connected = controller.is_connected
+        self._current_temp = None
+        self._setpoint_step = ih_device.get("farenheit_type")
+
+        self._max_temp = None
+        self._min_temp = None
+        self._target_temp = None
+
+        self._run_hours = None
+        self._rssi = None
+        self._swing = None
+        self._swing_list = None
+        self._vvane = None
+        self._hvane = None
+        self._power = False
+        self._fan_speed = STATE_UNKNOWN
+        self._hvac_mode = STATE_UNKNOWN
+        self._connection_retries = 0
+
+        self._hvac_modes = [
+            HVAC_MODE_HEAT_COOL,
+            HVAC_MODE_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_DRY,
+            HVAC_MODE_FAN_ONLY,
+            HVAC_MODE_OFF,
+        ]
+        self._fan_modes = controller.get_fan_speed_list(deviceid)
+
+        self._support = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+        if self._has_swing_control:
+            self._support |= SUPPORT_SWING_MODE
+            self._swing_list = [SWING_OFF, SWING_BOTH, SWING_VERTICAL, SWING_HORIZONTAL]
+
+        self._controller.add_update_callback(self.update_callback)
+
+    @property
+    def name(self):
+        """Return the name of the AC device."""
+        return self._devicename
+
+    @property
+    def temperature_unit(self):
+        """Intesishome API uses celsius on the backend."""
+        return TEMP_CELSIUS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attrs = {}
+        if self._has_swing_control:
+            attrs["vertical_vane"] = self._vvane
+            attrs["horizontal_vane"] = self._hvane
+
+        if self._controller.is_connected:
+            attrs["ha_update_type"] = "push"
+        else:
+            attrs["ha_update_type"] = "poll"
+
+        return attrs
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._deviceid
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            "identifiers": {(self._deviceid)},
+            "name": self._devicename,
+            "manufacturer": self._devicetype,
+        }
+
+    @property
+    def target_temperature_step(self) -> float:
+        """Return whether setpoint should be whole or half degree precision."""
+        if self._setpoint_step == 1:
+            return 1
+        return 0.5
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE)
+
+        if hvac_mode:
+            await self.async_set_hvac_mode(hvac_mode)
+
+        if temperature:
+            _LOGGER.debug("Setting %s to %s degrees", self._devicetype, temperature)
+            self._target_temp = temperature
```

Comment:
Normally either the update callback or `async_update` should update state holding instance attributes if the feedback from the device is fast enough. Then we don't need to update those attributes in the actuator methods.

What's the case here?

Reply:
There's about a 2 second delay from sending the temperature set command to getting the information back from the cloud controller. I'll move it into the library

Reply:
2 seconds is the timeout in the frontend. If the feedback takes longer than that its ok to update state in the actuator method.

Reply:
Looks like without this Home Assistant reverts back to the previous temperature value
Is it OK to call await async_update_ha_state(False) after self._target_temp = temperature? Otherwise it doesn't make any difference

Reply:
Yes, since the feedback is slow, it's ok.
But just call `self.async_write_ha_state()` instead. Add a comment about this too so it's clear to future readers.